### ISSUE TITLE
Empty Headers are not populated in milterSession headers

### ISSUE THREAD

**Issue by Ankit-Kulkarni** [phalaaxx/milter#23]
Hello , have observed that the milter doesn't pass around empty headers which causes an assumption of header being entirely absent and taking any action based on empty value is not possible. 

https://github.com/phalaaxx/milter/blob/1372e56163c24f063305614df58a029da60c7489/session.go#L180-L191

The following code above only populates the `m.headers` if HeaderData  is of len 2 . This eventually skips the cases where the header was present but with an empty value. 

```go
		if len(HeaderData) == 1 {
			m.headers.Add(HeaderData[0], "")
			// call and return milter handler
			return m.milter.Header(HeaderData[0], "", newModifier(m))
		}
```

The above code in addition to this should fix the case.

**Comment by phalaaxx** [comment]
Thanks, should be fixed now in 50ad4eb.

**Comment by Ankit-Kulkarni** [comment]
This fixed the issue thanks
